framework/src/media/utils/MediaUtils.cpp

```diff
@@ -22,19 +22,126 @@
+#define TS_PID(buffer)                          (((buffer[1] << 8) + buffer[2]) & BITS_MASK(13))
+#define TS_ADAPTION_FIELD_CONTROL(buffer)       ((buffer[3] >> 4) & BITS_MASK(2))
+#define TS_CONTINUITY_COUNTER(buffer)           ((buffer[3]) & BITS_MASK(4))
+#define TS_ADAPTION_FIELD_LENGTH(buffer)        (buffer[TS_HEAD_BYTES])
+
+#define SECTION_LENGTH(buffer)                  (((buffer[1] & 0x0F) << 8) + buffer[2])
+#define SECTION_HEAD_BYTES                      3       // table_id + ... + section_length
+#define PAT_PID                                 0x0000  // PID of Program Association Table
+#define NETWORK_PID_PN                          0x0000  // program number 0 specifies the network PID
+#define PMT_STREAMS_MAXIMUM                     10      // maximum number of streams in PMT
+#define PROGRAM_INFO_LENGTH(buffer)             (((buffer[10] << 8) + buffer[11]) & 0x0FFF)
+#define STREAM_INFO_HEAD_BYTES                  5       // each stream info head takes 5bytes
+#define ES_INFO_LENGTH(buffer)                  ((((buffer)[3] << 8) + (buffer)[4]) & 0x0FFF)
+#define CRC32_BYTES                             4       // CRC32 takes 4bytes
+
+// Supported audio stream types
+#define STREAM_TYPE_AUDIO_MPEG1                 0x03 // MPEG-1 audio (ISO/IEC 11172-3 audio)
+#define STREAM_TYPE_AUDIO_AAC                   0x0F // MPEG-2 lower bit-rate audio (ISO/IEC 13818-7 ADTS AAC)
+#define STREAM_TYPE_AUDIO_HE_AAC                0x11 // MPEG-4 LOAS multi-format framed audio (ISO/IEC 14496-3)
+
+#define PES_PACKET_HEAD_BYTES                   6   // packet_start_code_prefix + stream_id + PES_packet_length
+#define PES_STREAM_HEAD_BYTES                   3   // stream info + 7 flags + PES head data length fields
+#define PES_PACKET_START_CODE_PREFIX(buffer)    ((buffer[0] << 16) | (buffer[1] << 8) | buffer[2])
+#define PES_STREAM_ID(buffer)                   (buffer[3])
+#define PES_PACKET_LENGTH(buffer)               ((buffer[4] << 8) | buffer[5])
+#define PES_HEADER_DATA_LENGTH(buffer)          (buffer[PES_PACKET_HEAD_BYTES + 2])
+
+/****************************************************************************
+ * Private Declarations
+ ****************************************************************************/
+enum utils_error_e {
+	UTILS_ERROR_WANT_MORE_DATA = -6,
+	UTILS_ERROR_NOT_FOUND = -5,
+	UTILS_ERROR_NOT_SUPPORTED = -4,
+	UTILS_ERROR_NOT_AVAILABLE = -3,
+	UTILS_ERROR_BAD_PARAMETER = -2,
+	UTILS_ERROR_UNKNOWN = -1,
+	UTILS_ERROR_NONE = 0 /* No Error */
+};
+
+typedef struct {
+	uint16_t tsId;          // transport stream ID
+	uint16_t numOfPrgs;     // number of prgrams
+} psi_pat_info_t;
+
+typedef struct {
+	uint16_t prgNum;        // program number
+	uint16_t prgInfoLen;    // program info length, unused
+	uint16_t numOfStreams;  // number of streams
+} psi_pmt_info_t;
+
+typedef struct {
+	uint8_t type;           // stream type
+	uint16_t pid;           // elementary PID
+	uint16_t esInfoLen;     // ES info length, unused
```

Comment:
do you need to consider the bytes alignment for the structs defined in this file?

Reply:
Removed unnecessary members.